File: app/src/main/java/com/example/tah/ui/main/SectionsPagerAdapter.java
package com.example.tah.ui.main;

import androidx.annotation.NonNull;
import androidx.annotation.StringRes;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.example.tah.R;
import com.example.tah.ui.habit.HabitsFragment;
import com.example.tah.ui.task.TasksFragment;
import com.example.tah.ui.todo.TodosFragment;


public class SectionsPagerAdapter extends FragmentStateAdapter {

    @StringRes
    private static final int[] TAB_TITLES
            = new int[]{R.string.tab_tasks, R.string.tab_habits, R.string.tab_todos};

    public SectionsPagerAdapter(FragmentActivity fa){
        super(fa);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch(position){
            case 0:
                return new TasksFragment();
            case 1:
                return new HabitsFragment();
            default:
                return new TodosFragment();
        }
    }

    @Override
    public int getItemCount() {
        return TAB_TITLES.length;
    }

    public int getTabTitleId(int position){
        return TAB_TITLES[position];
    }
}

File: app/src/main/java/com/example/tah/ui/main/SectionsPagerAdapter.kt
package com.example.tah.ui.main

import androidx.annotation.StringRes
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.tah.R
import com.example.tah.ui.habit.HabitsFragment
import com.example.tah.ui.task.TasksFragment
import com.example.tah.ui.todo.TodosFragment

class SectionsPagerAdapter(fa: FragmentActivity): FragmentStateAdapter(fa) {

    @StringRes
    private val tabTitles = listOf(R.string.tab_tasks, R.string.tab_todos, R.string.tab_habits)

    override fun getItemCount(): Int = tabTitles.size

    override fun createFragment(position: Int): Fragment {
        return when(position){
            0 -> TasksFragment()
            1 -> TodosFragment()
            2 -> HabitsFragment()
            else -> TasksFragment()
        }
    }

    fun getTabTitleId(position: Int) = tabTitles[position]
}